src/transformers/models/propainter/modeling_propainter.py

```diff
@@ -0,0 +1,4630 @@
+                ),
+                nn.LeakyReLU(negative_slope, inplace=True),
+                nn.Conv2d(
+                    config.in_channels[0],
+                    config.num_channels,
+                    kernel_size=config.patch_size,
+                    stride=config.strides[1],
+                    padding=config.padding,
+                ),
+                nn.LeakyReLU(negative_slope, inplace=True),
+                nn.Conv2d(
+                    config.num_channels,
+                    config.num_channels * 2,
+                    kernel_size=config.patch_size,
+                    stride=1,
+                    padding=config.padding,
+                ),
+                nn.LeakyReLU(negative_slope, inplace=True),
+                nn.Conv2d(
+                    config.num_channels * 2,
+                    config.hidden_size - config.num_channels,
+                    kernel_size=config.patch_size,
+                    stride=1,
+                    padding=config.padding,
+                    groups=1,
+                ),
+                nn.LeakyReLU(negative_slope, inplace=True),
+                nn.Conv2d(
+                    config.hidden_size + config.num_channels,
+                    config.hidden_size,
+                    kernel_size=config.patch_size,
+                    stride=1,
+                    padding=config.padding,
+                    groups=2,
+                ),
+                nn.LeakyReLU(negative_slope, inplace=True),
+                nn.Conv2d(
+                    config.hidden_size + config.num_channels * 2,
+                    config.hidden_size - config.num_channels,
+                    kernel_size=config.patch_size,
+                    stride=1,
+                    padding=config.padding,
+                    groups=4,
+                ),
+                nn.LeakyReLU(negative_slope, inplace=True),
+                nn.Conv2d(
+                    config.hidden_size + config.num_channels,
+                    config.num_channels * 2,
+                    kernel_size=config.patch_size,
+                    stride=1,
+                    padding=config.padding,
+                    groups=8,
+                ),
+                nn.LeakyReLU(negative_slope, inplace=True),
+                nn.Conv2d(
+                    config.hidden_size,
+                    config.num_channels,
+                    kernel_size=config.patch_size,
+                    stride=config.strides[0],
+                    padding=config.padding,
+                    groups=1,
+                ),
+                nn.LeakyReLU(negative_slope, inplace=True),
+            ]
+        )
+
+    def forward(self, masked_inputs):
+        batch_size, _, _, _ = masked_inputs.size()
+        features = masked_inputs
+        for i, layer in enumerate(self.layers):
+            if i == 8:
+                x0 = features
+                _, _, height, width = x0.size()
+            if i > 8 and i % 2 == 0:
```

Comment:
maybe add a small comment here :sweat_smile: might not be intuitive to understand why there is this treatment of odd/even layers except at layer 8

Reply:
done